### PR TITLE
SALTO-4649: add important attribute core annotation

### PR DESCRIPTION
**Pull request by edenhassid**
add important attribute core annotation
the annotations are: `_important_values` and `_self_important_values`

---

_Additional context for reviewer_

---
_Release Notes_: 
core:
* add important attribute core annotation `_important_values` and `_self_important_values`

---
_User Notifications_: 
None

### CODE REVIEW

File: docs/syntax.md [hunk 1]
```diff
@@ -436,6 +437,22 @@ _alias = "Example Instance"
 }
 ```
 
+#### _important_values / _self_important_values
+
+This annotation is used to define a user-friendly alias for the element. The alias can be used in Salto enabled editors to display a shorter, clearer element name to users. Unlike the element ID, it does not have to be unique.
```
Comment (edenhassid) [inline docs/syntax.md:442]: need to change to be relevant for impartant values

Comment (edenhassid) [reply docs/syntax.md:442]: @liorn

File: docs/syntax.md [hunk 2]
```diff
@@ -293,6 +293,7 @@ Currently the following core annotations are supported:
 - [_updatable](#_changed_by)
 - [_deletable](#_changed_at)
 - [_additional_properties](#_additional_properties)
+- [_important_values / _self_important_values](#_important_values-_self_important_values)
```
Comment (ori-moisis) [inline docs/syntax.md:296]: does this link work? I think this is not the correct value to put there as the link location.
IIUC it should be
`#_important_values--_self_important_values`

seems like the other links with `/` in them also don't work. we should probably fix them as well

File: docs/syntax.md [hunk 3]
```diff
@@ -436,6 +437,28 @@ _alias = "Example Instance"
 }
 ```
 
+#### _important_values / _self_important_values
```
Comment (ori-moisis) [inline docs/syntax.md:440]: seems like some of the annotations have ` `` ` around them and others don't, may be worth fixing that and making sure everything looks in the same style
```suggestion
#### `_important_values` / `_self_important_values`
```

File: packages/adapter-api/src/builtins.ts [hunk 4]
```diff
@@ -103,6 +103,25 @@ const restrictionType = new ObjectType({
   },
 })
 
+const importantValueType = new ObjectType({
+  elemID: new ElemID(GLOBAL_ADAPTER, BUILTIN_TYPE_NAMES.IMPORTANT_VALUE),
+  fields: {
+    // eslint-disable-next-line camelcase
```
Comment (ori-moisis) [inline packages/adapter-api/src/builtins.ts:109]: why is this disable needed?

File: docs/syntax.md [hunk 5]
```diff
@@ -436,6 +437,28 @@ _alias = "Example Instance"
...
+
+This annotation is used to define what are these element's most important values. This list will be used in Salto enabled editors to highlight more important parts of the element, and index some of them for easy searchability. The list should include objects with the key of the element's important values and if it is indexed.
+Type: `string` Default: `undefined` Applicable to: Types
```
Comment (ori-moisis) [inline docs/syntax.md:443]: "Type: \`string`" is incorrect here

Comment (edenhassid) [reply docs/syntax.md:443]: what should the type be? is it ok to write `{value: string; indexed: boolean}[]` ?

Comment (ori-moisis) [reply docs/syntax.md]: you can see examples form other complex annotations like `_restriction`

**Review by ori-moisis — APPROVED** [review]

File: packages/adapter-api/src/constants.ts [hunk 6]
```diff
@@ -48,4 +50,5 @@ export const BUILTIN_TYPE_NAMES = {
   DEPENDENCY: 'dependency',
   HIDDEN_STRING: 'hidden_string',
   HIDDEN_BOOLEAN: 'hidden_boolean',
+  IMPORTANT_VALUE: 'importantValue',
```
Comment (netama) [inline packages/adapter-api/src/constants.ts:53]: shouldn't this have been `important_value`?

**Pull request merged**